gradle-incremental-configuration-cache/src/test/groovy/com/palantir/gradle/configcache/incremental/IncrementalConfigurationCacheTest.groovy

```diff
@@ -38,35 +39,59 @@ class IncrementalConfigurationCacheTest extends IntegrationTestKitSpec {
         """.stripIndent(true)
 
         expect:
-        assert runAndAssertNoConfigurationCache("classes")
+        def buildResult = createRunner(['classes', '--configuration-cache'] as String[]).buildAndFail()
+        assert buildResult.output.contains('Configuration cache allowed tasks file not found'),
+                "Expected task to blow up"
     }
 
-    def "tasks in allowlist run with config cache"() {
-        // Save the task graph of `build` to the allow list file
+    def "blows up if applied to non root project"() {
+        def subprojectDir = new File(getProjectDir(), 'subproject')
+        subprojectDir.mkdirs()
+        new File(subprojectDir, 'build.gradle') << """
+            apply plugin: 'com.palantir.incremental-configuration-cache'
+            apply plugin: 'java-library'
+        """.stripIndent(true)
```

Comment:
As discussed in person, you should be able to use IntegrationTestKitSpec's `file(...)`